### ISSUE TITLE
ARPACK wrapper 

### ISSUE THREAD

**Issue by TTitscher** [nutofem/nuto#17]
When activating ENABLE_ARPACK, FindArpack.cmake searches for a file "Arpack/Arpack.h". The `class EigenSolverArpack` includes it. This file was previously created by the [ARPACK.cmake.patch](https://sourceforge.net/p/nuto/wiki/Installation/) as a wrapper to the ARPACK fortran code. The CMake file also compiles a libArpack.a.

I naively included the "Arpack.h" (created by this patch) and linked the libarpack.a from the debian repository. Did not work. Help appreciated. :)

**Comment by TTitscher** [comment]
OK. Turns out I am an idiot and forgot how to properly link. 
~~~cpp
/* main.cpp */
#include "Arpack.h"
int main()
{
   return dnaupd_ (0,0,0,0,0, 0,0,0,0,0, 0,0,0,0,0, 0);
}
~~~
and
~~~
g++ main.cpp -L/usr/lib -larpack
~~~
compiles just fine. Trying to include that in NuTo now.

**Comment by TTitscher** [comment]
Fixed by https://github.com/nutofem/nuto/commit/97102c8aff8dfc183b0f2b49ac3d45f8ba69eece

**Comment by TTitscher** [comment]
There are still issues with the memory. Calculating the eigenvalues/eigenvectors in each iteration of a time integration scheme causes the memory to fill rapidly. 

Maybe switch to
https://eigen.tuxfamily.org/dox-devel/unsupported/group__ArpackSupport__Module.html?

Priority low, I guess...

**Comment by TTitscher** [comment]
We can remove our own implementation and use [`Eigen-unsupported`](https://eigen.tuxfamily.org/dox/unsupported/ArpackSelfAdjointEigenSolver_8h_source.html)

**Comment by Psirus** [comment]
Fixed by #202.
